Modules/socketmodule.c

```diff
@@ -1579,6 +1582,35 @@ makesockaddr(SOCKET_T sockfd, struct sockaddr *addr, size_t addrlen, int proto)
     }
 #endif /* HAVE_SOCKADDR_ALG */
 
+#ifdef AF_HYPERV
+    case AF_HYPERV:
+    {
+        SOCKADDR_HV *a = (SOCKADDR_HV *) addr;
+
+        wchar_t *guidStr;
+        PyObject *vmId;
+        PyObject *serviceId;
+
+        if (UuidToStringW(&a->VmId, &guidStr) == RPC_S_OUT_OF_MEMORY) {
+            PyErr_SetString(PyExc_MemoryError,
+                "out of memory in makesockaddr");
+            return NULL;
+        }
+        vmId = PyUnicode_FromWideChar(guidStr, -1);
+        RpcStringFreeW(&guidStr);
+
+        if (UuidToStringW(&a->ServiceId, &guidStr) == RPC_S_OUT_OF_MEMORY) {
+            PyErr_SetString(PyExc_MemoryError,
+                "out of memory in makesockaddr");
+            return NULL;
```

Comment:
```suggestion
            return PyErr_NoMemory();
```

Also, `vmId` leaks a reference here.

Also, intermingled declarations are allowed since C99. Please declare `vmId` when it is first used.